PayloadPkg/Library/PayloadLib/PayloadLib.c

```diff
@@ -360,3 +360,62 @@ GetLoaderStage (
 {
   return LOADER_STAGE_PAYLOAD;
 }
+
+/**
+  Get device address
+
+  @param[in]  DeviceType         The device type, refer OS_BOOT_MEDIUM_TYPE.
+  @param[in]  DeviceInstance     The device instance number starting from 0.
+  @param[in]  PciMmioAddr        Only applicable to PCI device. TRUE will return
+                                 PCI MMIO address, else return PCI BDF.
+
+  @retval     Device address for a given device instance, return 0 if the device
+              could not be found from device table.
+**/
+UINT32
+EFIAPI
+GetDeviceAddr (
```

Comment:
Can we share the code with Stage ?   Maybe move it to common.   Only need to implement GetDeviceTable() API differently ?

Reply:
OK. will update it.